core/store/src/cold_storage.rs

```diff
@@ -187,6 +202,39 @@ fn get_keys_from_store(
                     )?;
                     keys
                 }
+                DBKeyType::TransactionHash => chunks
+                    .iter()
+                    .flat_map(|c| c.transactions().iter().map(|t| t.get_hash().as_bytes().to_vec()))
+                    .collect(),
+                DBKeyType::ReceiptHash => chunks
+                    .iter()
+                    .flat_map(|c| c.receipts().iter().map(|r| r.get_hash().as_bytes().to_vec()))
+                    .collect(),
+                DBKeyType::OutcomeId => {
+                    let mut outcomes: Vec<Vec<CryptoHash>> = vec![];
+                    for shard_id in 0..shard_layout.num_shards() {
+                        debug_assert_eq!(
+                            DBCol::OutcomeIds.key_type(),
+                            &[DBKeyType::BlockHash, DBKeyType::ShardId]
+                        );
+
+                        outcomes.push(
+                            store
+                                .get_ser(
+                                    DBCol::OutcomeIds,
+                                    &join_two_keys(
+                                        &block_hash_key,
+                                        &shard_id.to_le_bytes().to_vec(),
+                                    ),
+                                )?
+                                .unwrap_or_default(),
+                        );
+                    }
+                    outcomes
+                        .iter()
+                        .flat_map(|os| os.iter().map(|o| o.as_bytes().to_vec()))
+                        .collect()
```

Comment:
No need to do `to_vec` in `&shard_id.to_le_bytes().to_vec()`.  Furthermore,

```suggestion
                    debug_assert_eq!(
                        DBCol::OutcomeIds.key_type(),
                        &[DBKeyType::BlockHash, DBKeyType::ShardId]
                    );
                    (0..shard_layout.num_shards())
                        .map(|shard_id| {
                            Ok(store
                               .get_ser::<Vec<CryptoHash>>(
                                   DBCol::OutcomeIds,
                                   &join_two_keys(
                                       &block_hash_key,
                                       &shard_id.to_le_bytes(),
                                   ),
                               )?
                               .unwrap_or_default())
                        })
                        .collect::<Result<Vec<_>, _>()?
                        .into_iter()
                        .flat_map(|hashes| hashes.into_iter().map(|hash| hash.as_bytes().to_vec()))
                        .collect()
```

Might need to add some more types somewhere but should work.